llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeWinogradPass.cpp

```diff
@@ -156,59 +151,134 @@ class ReifyWinogradInputTransform final
-    offsets = SmallVector<OpFoldResult>(numImageDims, zero);
-    offsets.append(ivs.begin(), ivs.end());
-    sizes = SmallVector<OpFoldResult>(inputOp.getOutputOperandRank(), one);
-    sizes[0] = sizes[1] = inputTileSizeAttr;
+    auto stridesOutputSlice = SmallVector<OpFoldResult>(inputOp.getOutputOperandRank(), one);
+    auto offsetsOutputSlice = SmallVector<OpFoldResult>(numImageDims, zero);
+    offsetsOutputSlice.append(ivs.begin(), ivs.end());
+    auto sizesOutputSlice = SmallVector<OpFoldResult>(inputOp.getOutputOperandRank(), one);
+    sizesOutputSlice[0] = sizesOutputSlice[1] = inputTileSizeAttr;
     tensorType = RankedTensorType::get(inputTileSquare, elementType);
     Value iterArg = loopNest.loops.back().getRegionIterArg(0);
     Value outputSlice = rewriter.create<tensor::ExtractSliceOp>(
-        loc, tensorType, iterArg, offsets, sizes, strides);
+        loc, tensorType, iterArg, offsetsOutputSlice, sizesOutputSlice, stridesOutputSlice);
 
-    // Create computation
-    Value result, AMatrix, BMatrix;
-    linalg::MatmulOp matmulOp;
-    for (int i = 0; i < 2; i++) {
-      fillOp = rewriter.create<linalg::FillOp>(loc, ValueRange{zeroF32},
-                                               ValueRange{outputSlice});
-      if (i == 0) {
-        AMatrix = inputSlice;
-        BMatrix = BV;
-      } else {
-        AMatrix = BTV;
-        BMatrix = result;
-      }
-      matmulOp = rewriter.create<linalg::MatmulOp>(
-          loc, tensorType, ValueRange{AMatrix, BMatrix}, fillOp.result());
-      result = matmulOp.getResult(0);
-    }
+    IntegerAttr outputTileSizeI64Attr = rewriter.getI64IntegerAttr(inputOp.getOutputTileSize());
+    IntegerAttr kernelSizeI64Attr = rewriter.getI64IntegerAttr(inputOp.getKernelSize());
+    DenseI64ArrayAttr imageDimensionsDenseI64ArrayAttr = rewriter.getDenseI64ArrayAttr(inputOp.imageDimensions());
+    tiledWinogradInputTransformOp = rewriter.create<WinogradInputTransformOp>(
+      inputOp.getLoc(),
+      tensorType, dynamicSlice, outputSlice, outputTileSizeI64Attr, kernelSizeI64Attr, imageDimensionsDenseI64ArrayAttr);
 
     // Insert results into output slice
     Value updatedOutput = rewriter.create<tensor::InsertSliceOp>(
-        loc, result, iterArg, offsets, sizes, strides);
+        loc, tiledWinogradInputTransformOp.getResult()[0], iterArg, offsetsOutputSlice, sizesOutputSlice, stridesOutputSlice);
 
     // Replace returned value
     if (scf::YieldOp yieldOp = dyn_cast<scf::YieldOp>(
             loopNest.loops.back().getBody()->getTerminator())) {
+      OpBuilder::InsertionGuard yieldGuard(rewriter);
+      rewriter.setInsertionPoint(yieldOp);
       rewriter.replaceOpWithNewOp<scf::YieldOp>(yieldOp, updatedOutput);
     }
     inputOp.getResults()[0].replaceAllUsesWith(loopNest.results[0]);
+
     return success();
   }
-};
+// };
+
+static void decomposeTiledWinogradInputTransformOp(WinogradInputTransformOp tiledWinogradInputTransformOp,
```

Comment:
Its good practice to just return `LogicalResult` here instead of void so that you can thred through errors.